src/main/kotlin/org/rust/ide/refactoring/implementMembers/impl.kt

```diff
@@ -104,9 +105,15 @@ private fun insertNewTraitMembers(
             val whitespaces = createExtraWhitespacesAroundFunction(addedMember, next)
             existingMembers.addAfter(whitespaces, addedMember)
         }
+
+        if (addedMember is RsFunction && templateMembers.isEmpty())
+        {
+            templateMembers.add(addedMember)
+        }
     }
 
     importTypeReferencesFromElements(existingMembers, selected, trait.subst)
+    editor?.buildAndRunTemplate(existingMembers, templateMembers.mapNotNull { it.block?.expr?.createSmartPointer() })
```

Comment:
Running template builder here looks redundant.
At least, I don't expect it after using support for other languages (kotlin, java) where only body of the first method is selected (without template)
So let's make the same behavior as implemented in other languages

Reply:
Also, it can be easily tested

Reply:
Can you point me to an API that I can use to select the body of the first method? In the first commit of this PR I simply selected the `unimplemented!` body of the first method. However, if I simply select it and the user presses Enter, it will delete the `unimplemented!` text, instead of leaving it in place, which is wrong.

Reply:
@Kobzol why do you think that it's wrong?
It's exactly the same behavior that as all other languages provide

![2020-01-27 18 07 19](https://user-images.githubusercontent.com/2539310/73185881-ee917580-412f-11ea-8bfe-b43f4fd2f78f.gif)

Reply:
Ok.. it seemed a bit unintuitive to me, since with the template builder Enter keeps the original text there :) In that case I'll just revert to the first commit and add tests.